firebase-appdistribution/test-app/src/main/java/com/googletest/firebase/appdistribution/testapp/MainActivity.kt

```diff
@@ -26,108 +34,200 @@ class MainActivity : AppCompatActivity() {
             "Sample App v${BuildConfig.VERSION_NAME} (${BuildConfig.VERSION_CODE})"
         setupUI(isSignedIn = firebaseAppDistribution.isTesterSignedIn, isUpdateAvailable = false)
 
-        /** Basic Configuration */
-        findViewById<AppCompatButton>(R.id.sign_in_button).visibility = View.GONE
-        findViewById<AppCompatButton>(R.id.check_for_update).visibility = View.GONE
-        findViewById<AppCompatButton>(R.id.sign_out).visibility = View.GONE
+        val progressBar = findViewById<ProgressBar>(R.id.progress_bar)
+        val progressPercent = findViewById<TextView>(R.id.progress_percentage)
+        val signInButton = findViewById<AppCompatButton>(R.id.sign_in_button)
+        val signOutButton = findViewById<AppCompatButton>(R.id.sign_out)
+        val checkForUpdateButton = findViewById<AppCompatButton>(R.id.check_for_update)
+        val updateAppButton = findViewById<AppCompatButton>(R.id.update_app)
+        val updateIfNewReleaseAvailableButton = findViewById<AppCompatButton>(R.id.basic_config)
+        val updateIfNewReleaseAvailableButtonBackground = findViewById<AppCompatButton>(R.id.basic_config2)
+        val signInButtonBackground = findViewById<AppCompatButton>(R.id.sign_in_button2)
+        val signOutButtonBackground = findViewById<AppCompatButton>(R.id.sign_out2)
+        val checkForUpdateButtonBackground = findViewById<AppCompatButton>(R.id.check_for_update2)
+        val updateAppButtonBackground = findViewById<AppCompatButton>(R.id.update_app2)
+
+        /** Button Listeners */
+        updateIfNewReleaseAvailableButton.setOnClickListener {
+                   if (updateTask == null || updateTask?.isComplete == true) {
+           updateTask = firebaseAppDistribution.updateIfNewReleaseAvailable().addOnFailureListener {
+               val ex = it as FirebaseAppDistributionException1
+               Log.d("FirebaseAppDistribution", "MAINACTIVITY:ERROR ERROR. CODE: " + it.errorCode)
```

Comment:
Could we extract this failure listener and add it to all the buttons for testing?